code/messaging/src/phone/java/com/adobe/marketing/mobile/Messaging.java

```diff
@@ -390,12 +447,26 @@ public void call(final Event event) {
      *     propositions
      */
     public static void updatePropositionsForSurfaces(@NonNull final List<Surface> surfaces) {
-        if (surfaces == null || surfaces.isEmpty()) {
+        updatePropositionsForSurfaces(surfaces, null);
+    }
+
+    /**
+     * Dispatches an event to fetch propositions for the provided surfaces from Adobe Journey
+     * Optimizer via the Experience Edge network.
+     *
+     * @param surfaces A {@code List<Surface>} containing {@link Surface}s to be used for retrieving
+     *     propositions
+     * @param callback An optional callback to be called once the proposition response has been
+     *     processed by the Messaging extension
+     */
+    public static void updatePropositionsForSurfaces(
+            @NonNull final List<Surface> surfaces,
+            @Nullable final AdobeCallback<Boolean> callback) {
+        if (surfaces.isEmpty()) {
```

Comment:
nit: In Java the @NonNull annotation is not really enforced. The compiler will give a warning if this API is called with a null value but the build will not fail. To cover our bases, is it ok if I add back the null check?